### PR TITLE
Add :safe clause for inf-clojure-repl-use-same-window

### PR DESCRIPTION
**Pull request by arichiardi**
- [X] The commits are consistent with our [contribution guidelines][1]

### CODE REVIEW

File: inf-clojure.el [hunk 1]
```diff
@@ -367,6 +367,7 @@ This should usually be a combination of `inf-clojure-prompt' and
   "Controls whether to display the REPL buffer in the current window or not."
   :type '(choice (const :tag "same" t)
                  (const :tag "different" nil))
+  :safe (lambda (s) (or (string-equal "same" s) (string-equal "different" s)))
```
Comment (bbatsov) [inline inf-clojure.el]: The actual values are `t` and `nil`, not "same" and "different". A guess this should just be a `boolean?`.

**Comment by arichiardi** [comment]
Oh, well it is working this way, emacs does no show it as unsafe, maybe becuaes a string is `t`?

**Comment by bbatsov** [comment]
I think you should just use `booleanp` here.

**Comment by bbatsov** [comment]
Here's an example - https://github.com/clojure-emacs/cider/blob/976ffdbb57bf1a394c4705bdf0ed773c054c63bc/cider-eldoc.el#L85

**Comment by arichiardi** [comment]
Will do that

**Pull request merged**